include/config.h

```diff
@@ -40,6 +40,8 @@ namespace detail {
 		 */
 		const std::optional<device_config>& get_device_config() const { return device_cfg; };
 		std::optional<bool> get_enable_device_profiling() const { return enable_device_profiling; };
+		std::optional<bool> is_dry_run() const { return dry_run; };
```

Comment:
I think we could just default to `false` if it is not set and void the `optional`.